ITISFoundation/osparc-simcore
# ⬆️ Maintenance: upgrade dependencies of webserver service

## What do these changes do?

Before adding new features in the web-server we proceed with a prune and upgrade of the dependencies:

- ⬆️ upgrades all dependencies of webserver
- 🗑️ replaced ``semantic_version`` by ``packaging.version`` already installed
- 🔨 enhanced ``requirements/tools/check_changes.py`` script

###  Highlights on updated libraries (only updated libraries are included)

- #packages before: 21
- #packages after : 20

|#|name|before|after|upgrade|count|packages|
|-|----|------|-----|-------|-----|--------|
|   1 | aio-pika                  | 7.1.2           | 7.2.0      | *minor*    | 1 | web⬆️ |
|   2 | click                     | 8.0.4           | 8.1.3      | *minor*    | 3 | web⬆️🧪🔧 |
|   3 | [cryptography](https://cryptography.io/en/latest/changelog/#main)              | 36.0.2          | 37.0.2     | **MAJOR**  | 2 | web⬆️🧪 |
|   4 | email-validator           | 1.1.3           | 1.2.1      | *minor*    | 1 | web⬆️ |
|   5 | faker                     | 13.7.0          | 13.11.0    | *minor*    | 1 | web🧪 |
|   6 | hypothesis                | 6.46.2          | 6.46.3     |            | 1 | web🧪 |
|   7 | jinja2                    | 3.1.1           | 3.1.2      |            | 1 | web⬆️ |
|   8 | [jsondiff](https://github.com/xlwings/jsondiff/releases)                  | 1.3.1           | 2.0.0      | **MAJOR**  | 1 | web⬆️ |
|   9 | orjson                    | 3.6.7           | 3.6.8      |            | 1 | web⬆️ |
|  10 | pint                      | 0.18            | 0.19.2     | *minor*    | 1 | web⬆️ |
|  11 | prometheus-client         | 0.13.1          | 0.14.1     | *minor*    | 1 | web⬆️ |
|  12 | pyparsing                 | 3.0.7           | 3.0.9      |            | 2 | web⬆️🧪 |
|  13 | redis                     | 4.2.2           | 4.3.1      | *minor*    | 2 | web⬆️🧪 |
|  14 | semantic-version          | 2.9.0           | 🗑️ removed |  | 1 | web⬆️ |
|  15 | setproctitle              | 1.2.2           | 1.2.3      |            | 1 | web⬆️ |
|  16 | sqlalchemy                | 1.4.32          | 1.4.36     |            | 2 | web⬆️🧪 |
|  17 | tqdm                      | 4.63.1          | 4.64.0     | *minor*    | 1 | web⬆️ |
|  18 | typing-extensions         | 4.1.1           | 4.2.0      | *minor*    | 3 | web⬆️🧪🔧 |
|  19 | ujson                     | 5.1.0           | 5.2.0      | *minor*    | 1 | web⬆️ |
|  20 | werkzeug                  | 2.0.3           | 2.1.2      | *minor*    | 1 | web⬆️ |
|  21 | wrapt                     | 1.14.0          | 1.14.1     |            | 2 | web⬆️🧪 |

*Legend*: 
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

## Related issue/s

- maintenance

## How to test

To test that all requirements are frozen:
1. ``cd web/server``
1. install for production: ``make install-prod``
1. run recipe ``make pip-freeze``  -> ``freeze-base.ignore.txt`` and ``freeze-now.ignore.txt``
1. compare those two files
1. should only differ in : 
   - ``simcore-*`` libraries
   - capitalization or ``-`` vs ``_``
   - ``freeze-now`` include ``pip-tools`` and ``pep517``  tooling (for ``pip-sync``)
